lib/bundler/injector.rb

```diff
@@ -90,5 +115,147 @@ def append_to(gemfile_path, new_gem_lines)
         f.puts new_gem_lines
       end
     end
+
+    # evalutes a gemfile to remove the specified gem
+    # from it.
+    def remove_deps(gemfile_path)
+      # get initial snap shot of the gemfile
+      initial_gemfile = IO.readlines(gemfile_path)
+
+      # inform user of the the gemfile currently
+      # being evaluated
+      Bundler.ui.info "Removing gems from #{gemfile_path}"
+
+      # evaluate the Gemfile we have
+      builder = Dsl.new
+      builder.eval_gemfile(gemfile_path)
+
+      # remove gems from dependencies
+      removed_deps = remove_gems_from_dependencies(builder, @deps)
```

Comment:
this causes the following output, which is not quite right:

```ruby
# Gemfile
eval_gemfile 'Gemfile2'
gem 'ab'
# Gemfile2
gem("ab")
```

=>

```
Removing gems from Gemfile
ab could not be removed.
No gems were removed from the gemfile.
Removing gems from Gemfile2
ab (>= 0) was removed.
```

Reply:
Looking into this.

Reply:
How about if we add a method `gemfile` to `Dependency` that will hold the gemfile to which the gem belongs? In that way, we can easily filter out correct gems.

Reply:
Added the `gemfile` [here](https://github.com/bundler/bundler/pull/6513/commits/d16ad19f60ba42adf880da7fabafe8308a40b1a9)